argusclient/client.py

```diff
@@ -409,12 +439,74 @@ def get_user_dashboards(self, ownerName=None, shared=True, limit=None, version=N
         """
         return self.argus._request("get", "dashboards", params=dict(owner=ownerName, shared=shared, limit=limit, version=version))
 
+
+class GroupPermissionsServiceClient(BaseUpdatableModelServiceClient):
+    """
+    Service class that interfaces with the Argus grouppermissions endpoint.
+
+    There is no need to instantiate this directly, as it is available as :attr:`argusclient.client.ArgusServiceClient.grouppermissions` attribute.
+    """
+
+    def __init__(self, argus, get_all_req_opts=None):
+        """
+      :param get_all_req_opts: See BaseModelServiceClient.__init__() for description.
+      """
+        if not get_all_req_opts:
+            get_all_req_opts = {}
+        get_all_req_opts.setdefault(REQ_PATH, "grouppermission")
+        super(GroupPermissionsServiceClient, self).__init__(GroupPermission, argus, id_path="grouppermission/",
+                                                            get_all_req_opts=get_all_req_opts)
+
+    def _init_all(self, coll=None):
+        if not self.get_all_req_opts.get(REQ_PATH):
+            raise TypeError("Unsupported operation on: %s" % type(self))
+        if not self._retrieved_all:
+            resp = convert(self.argus._request(self.get_all_req_opts.get(REQ_METHOD, "get"),
+                                               self.get_all_req_opts.get(REQ_PATH, None),
+                                               params=self.get_all_req_opts.get(REQ_PARAMS, None),
+                                               dataObj=self.get_all_req_opts.get(REQ_BODY, None)))
+            for group_id, perms in resp.items():
+                self._coll[group_id] = perms
+            self._retrieved_all = True
+
+
+    def get_permissions_for_group(self, groupId):
+        return convert(self.argus._request("get", "grouppermission", params=list(groupId)))
+
+    def add_permissions_for_group(self, grouppermission):
+        """
+               Updates permissions in groups
+               :return: a list of :class:`argusclient.model.GroupPermission` objects with all fields populated.
+        """
+        if not isinstance(grouppermission, GroupPermission):
+            raise TypeError("Need a GroupPermission object, got: %s" % type(grouppermission))
+        updatedGroup = convert(self.argus._request("post", "grouppermission", dataObj=grouppermission))
```

Comment:
@sanjana-sfdc , any idea why this would be returning permission object instead of groupPermission? This is happening in the test_group file